### DISCUSSION TITLE
Change name of EFI file generated (zfsbootmenu.efi)

### DISCUSSION THREAD

**Discussion by reefland** [zbm-dev/zfsbootmenu#369] [Questions] [answered]
I have new hardware [Minisform UM560](https://store.minisforum.com/products/um560) - AMD Ryzen 5 5625U, 6 Cores/12 Threads based.

After days of troubleshooting, I've figured out the UEFI BIOS is brain dead.  It is hard coded to look for a file path name of `EFI/boot/bootx64.efi`.  Once I manually made a copy of `EFI/boot/zfsbootmenu.efi` to `EFI/boot/bootx64.efi` the system was able to boot.  Without this it acted as if no ESPs existed and went straight into UEFI BIOS Setup.

Manually copying this file doesn't seem like the right way to handle this, and the `zfsbootmenu/config.yaml` didn't seem to have an entry to set the filename created.

What is the proper procedure to have ZBM create a `bootx64.efi` instead of `zfsbootmenu.efi`  when manually building locally?  (I know I could curl the EFI directly and just place it with the alternate filename).

**Comment by ahesford** [comment, marked as answer]
In `config.yaml`, make sure you have the following options (add any missing keys, replace any conflicts; this assumes your EFI system partition is mounted at `/boot/efi`):

```yaml
EFI:
  Versions: false
  ImageDir: /boot/efi/EFI/boot
Kernel:
  Prefix: bootx64
```

**Reply by ahesford** [reply]
Also, if you want totally arbitrary flexibility, you can define

```yaml
Global:
  PostHooksDir: /etc/zfsbootmenu/generate-zbm.post.d
```

Make the `/etc/zfsbootmenu/generate-zbm.post.d` directory and dump any executable shell scripts (or other programs) in there; `generate-zbm` will run these scripts after producing and saving the image to the configured directory, so you can copy things around however you want.

**Comment by reefland** [comment]
Thank you for the prompt response, appreciated.
